tests/components/modbus/test_light.py

```diff
@@ -352,3 +384,199 @@ async def test_no_discovery_info_light(
     )
     await hass.async_block_till_done()
     assert LIGHT_DOMAIN in hass.config.components
+
+
+@pytest.mark.parametrize(
+    "do_config",
+    [
+        {
+            CONF_LIGHTS: [
+                {
+                    CONF_NAME: TEST_ENTITY_NAME,
+                    CONF_ADDRESS: 1234,
+                    CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                    CONF_SCAN_INTERVAL: 0,
+                    CONF_BRIGHTNESS_REGISTER: 1,
+                }
+            ]
+        }
+    ],
+)
+async def test_brightness_light(hass: HomeAssistant, mock_modbus_ha) -> None:
+    """Test Modbus Light brightness."""
+    assert hass.states.get(ENTITY_ID).state == STATE_OFF
+
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_ON,
+        service_data={
+            ATTR_ENTITY_ID: ENTITY_ID,
+            ATTR_BRIGHTNESS: 128,
+        },
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    assert hass.states.get(ENTITY_ID).state == STATE_ON
+
+    calls = mock_modbus_ha.write_register.call_args_list
+    modbus_brightness = ModbusLight._convert_brightness_to_modbus(brightness=128)
+
+    assert any(
+        call.args[0] == 1 and call.kwargs["value"] == modbus_brightness
+        for call in calls
+    )
+
+    await hass.services.async_call(
+        LIGHT_DOMAIN,
+        SERVICE_TURN_OFF,
+        service_data={ATTR_ENTITY_ID: ENTITY_ID},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    assert hass.states.get(ENTITY_ID).state == STATE_OFF
+
+
+@pytest.mark.parametrize(
+    "do_config",
+    [
+        {
+            CONF_LIGHTS: [
+                {
+                    CONF_NAME: TEST_ENTITY_NAME,
+                    CONF_ADDRESS: 1234,
+                    CONF_WRITE_TYPE: CALL_TYPE_REGISTER_HOLDING,
+                    CONF_SCAN_INTERVAL: 0,
+                    CONF_BRIGHTNESS_REGISTER: 1,
+                    CONF_COLOR_TEMP_REGISTER: 2,
+                }
+            ]
+        }
+    ],
+)
+async def test_color_temp_light(hass: HomeAssistant, mock_modbus_ha) -> None:
```

Comment:
we could combine this test with the brightness one with test params. each param tuple would have both the service data for `turn_on` (which can include only temp, only brightness, or both) and the expected results.